docker/test/fasttest/Dockerfile

```diff
@@ -44,42 +44,18 @@ RUN sed -i '/_IMPORT_CHECK_FILES_FOR_\(mlir-\|llvm-bolt\|merge-fdata\|MLIR\)/ {s
 # It's very dirty workaround, better to build compiler and LLVM ourself and use it. Details: https://github.com/llvm/llvm-project/issues/95792
 RUN test ! -d /usr/lib/llvm-18/lib/clang/18/lib/x86_64-pc-linux-gnu || ln -s /usr/lib/llvm-18/lib/clang/18/lib/x86_64-pc-linux-gnu /usr/lib/llvm-18/lib/clang/18/lib/x86_64-unknown-linux-gnu
 
-ARG CCACHE_VERSION=4.6.1
-RUN mkdir /tmp/ccache \
-    && cd /tmp/ccache \
-    && curl -L \
-        -O https://github.com/ccache/ccache/releases/download/v$CCACHE_VERSION/ccache-$CCACHE_VERSION.tar.xz \
-        -O https://github.com/ccache/ccache/releases/download/v$CCACHE_VERSION/ccache-$CCACHE_VERSION.tar.xz.asc \
-    && gpg --recv-keys --keyserver hkps://keyserver.ubuntu.com 5A939A71A46792CF57866A51996DDA075594ADB8 \
-    && gpg --verify ccache-4.6.1.tar.xz.asc \
-    && tar xf ccache-$CCACHE_VERSION.tar.xz \
-    && cd /tmp/ccache/ccache-$CCACHE_VERSION \
-    && cmake -DCMAKE_INSTALL_PREFIX=/usr \
-        -DCMAKE_BUILD_TYPE=None \
-        -DZSTD_FROM_INTERNET=ON \
-        -DREDIS_STORAGE_BACKEND=OFF \
-        -Wno-dev \
-        -B build \
-        -S . \
-    && make VERBOSE=1 -C build \
-    && make install -C build \
-    && cd / \
-    && rm -rf /tmp/ccache
-
-ARG TARGETARCH
-ARG SCCACHE_VERSION=v0.7.7
-ENV SCCACHE_IGNORE_SERVER_IO_ERROR=1
-# sccache requires a value for the region. So by default we use The Default Region
-ENV SCCACHE_REGION=us-east-1
-RUN arch=${TARGETARCH:-amd64} \
-  && case $arch in \
-    amd64) rarch=x86_64 ;; \
-    arm64) rarch=aarch64 ;; \
-  esac \
-  && curl -Ls "https://github.com/mozilla/sccache/releases/download/$SCCACHE_VERSION/sccache-$SCCACHE_VERSION-$rarch-unknown-linux-musl.tar.gz" | \
-    tar xz -C /tmp \
-  && mv "/tmp/sccache-$SCCACHE_VERSION-$rarch-unknown-linux-musl/sccache" /usr/bin \
-  && rm "/tmp/sccache-$SCCACHE_VERSION-$rarch-unknown-linux-musl" -r
+# Rust toolchain and libraries
+ENV RUSTUP_HOME=/rust/rustup
+ENV CARGO_HOME=/rust/cargo
+ENV PATH="/rust/cargo/bin:${PATH}"
+RUN curl https://sh.rustup.rs -sSf | bash -s -- -y && \
+    chmod 777 -R /rust && \
+    rustup toolchain install nightly-2024-11-01 && \
```

Comment:
AFAICT this is needed because we use some deps that require a newer nightly than we use (`nightly-2024-04-01`). This means we'll need to upgrade the toolchain instructions and vendored rust deps.